### PR TITLE
Use Parg to self-doc vpm actions and help

### PR DESCRIPTION
**Pull request by dhalman**

### CODE REVIEW

File: utils.go [hunk 1]
```diff
@@ -47,3 +30,41 @@ func removeBranchHash(gitURL string) (out string) {
...
+	} else {
+		msg = strings.Join(args, ", ")
+	}
+
+	return
+}
+
+func showHelp(cmd *flag.Command) {
+	if cmd == nil {
+		fmt.Println(flag.Help())
```
Comment (itsmontoya) [inline utils.go]: This is the more idiomatic way to approach this for go

```go
if cmd == nil {
    fmt.Println(flag.Help()
    return
}

fmt.Println(cmd.Help())
```

**Review by itsmontoya — DISMISSED** [review]
Can you adjust the if/else blocks to be if/return blocks? :)

File: command.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+	fmt.Println(cmd.Help())
+	return
+}
+
+func doc(cmd *flag.Command) (err error) {
+	_, msg := commandParams(cmd)
+	out.Notificationf("Documenting %s...", msg)
+
+	/*
+		if err := v.documentPlugins(args...); err != nil {
```
Comment (dhalman) [inline command.go]: Getting started on this now :)

File: main.go [hunk 3]
```diff
@@ -27,58 +27,22 @@ func main() {
...
-
-		v.listPlugins(args...)
-
-	case "help":
-		// TODO: Use parg for help docs?
-		out.Notification("Supported commands are: update, build, list, and help.")
-
-	default:
-		err = fmt.Errorf("invalid command, \"%s\" is not supported", cmd)
+	var cmd *flag.Command
```
Comment (dhalman) [inline main.go:39]: @itsmontoya  :) :) :)

Comment (dhalman) [reply main.go:39]: Told you the switch would go away!

File: vpm.go [hunk 4]
```diff
@@ -29,7 +29,12 @@ func (v *vpm) getPluginsMatchingAny(pluginNames ...string) (plugins []string) {
 }
 
 func (v *vpm) addPlugins(pluginNames ...string) (err error) {
-	for _, pluginKey := range v.getPluginsMatchingAny(pluginNames...) {
+	plugins := v.getPluginsMatchingAny(pluginNames...)
+	if len(plugins) == 0 {
+		return fmt.Errorf("No plugin found matching %v", pluginNames)
+	}
+
+	for _, pluginKey := range plugins {
```
Comment (dhalman) [inline vpm.go:37]: Throw an error if no plugins match, this prevents a typo causing a failure to update and giving similar output

**Review by itsmontoya — APPROVED** [review]

**Pull request merged**